Antidote/CallControlsView.m

```diff
@@ -130,11 +168,57 @@ - (void)resumeButtonPressed
     [self.delegate callControlsResumeButtonPressed:self];
 }
 
-#pragma makr - Public
+- (void)videoButtonPressed
+{
+    [self.delegate callControlsVideoButtonPressed:self];
+}
+
+- (void)endCallButtonPressed
+{
+    [self.delegate callControlsEndCallButtonPressed:self];
+}
+
+#pragma mark - CompactControlsViewDelegate
+
+- (void)compactControlsVideoButtonPressed:(CompactControlsView *)compactControlsView
+{
+    [self.delegate callControlsVideoButtonPressed:self];
+}
+
+- (void)compactControlsMicButtonPressed:(CompactControlsView *)compactControlsView
+{
+    [self.delegate callControlsMicButtonPressed:self];
+}
+
+- (void)compactControlsSpeakerButtonPressed:(CompactControlsView *)compactControlsView
+{
+    [self.delegate callControlsSpeakerButtonPressed:self];
+}
+
+- (void)compactControlsEndCallButtonPressed:(CompactControlsView *)compactControlsView
+{
+    [self.delegate callControlsEndCallButtonPressed:self];
+}
+
+- (void)compactControlsResumeButtonPressed:(CompactControlsView *)compactControlsView
+{
+    [self.delegate callControlsResumeButtonPressed:self];
+}
+
+#pragma mark - Public
+
+- (void)setType:(CallsControlsViewType)type
+{
+    _type = type;
+
+    self.compactView.hidden = (type == CallsControlsViewTypeExpand);
+    self.expandedView.hidden = (type == CallsControlsViewTypeCompact);
```

Comment:
Despite the fact that compact view will move to controller, it would be better to do
```
self.compactView.hidden = (type != CallsControlsViewTypeCompact);
self.expandedView.hidden = (type != CallsControlsViewTypeExpand);
```

That way there will be no errors if we'll add more types.